lib/raygun.ts

```diff
@@ -337,19 +337,32 @@ class Raygun {
       return;
     }
 
+    // Handles uncaught exceptions, e.g. missing try/catch on a throw.
     process.on("uncaughtExceptionMonitor", (e) => {
-      this.sendSync(e);
+      this.sendSync(e, ["uncaughtException"]);
+    });
+
+    // Handles uncaught Promise rejections, e.g. missing .catch(...) on a Promise.
+    // Unhandled Promise rejections are not caught by the "uncaughtExceptionMonitor".
+    process.on("unhandledRejection", (reason, promise) => {
+      if (reason instanceof Error || typeof reason === "string") {
+        this.sendSync(reason, ["unhandledRejection"]);
+      } else {
+        // `reason` type is unknown, wrap in String
+        this.sendSync(`Unhandled Rejection: ${reason}`, ["unhandledRejection"]);
+      }
     });
   }
 
   /**
    * Send error using synchronous transport.
    * Only used to report uncaught exceptions.
```

Comment:
Tiny: Should we add unhandled promises here? Otherwise it sounds like we're specifically excluding them.

Reply:
The function is internal only but good to keep documentation consistent

Reply:
Updated jsdoc comment